internal/querynode/shard_cluster.go

```diff
@@ -1031,11 +1031,10 @@ func (sc *ShardCluster) Search(ctx context.Context, req *querypb.SearchRequest,
 		resultMut.Lock()
 		defer resultMut.Unlock()
 		if streamErr != nil {
-			cancel()
-			// not set cancel error
-			if !errors.Is(streamErr, context.Canceled) {
+			if err == nil {
```

Comment:
maybe wrapping errors is better, so we can find out all errors

Reply:
first error may be canceled in this case, query one shard failed and cancel the whole search request for the the shard leader